go_modules/lib/dependabot/go_modules/update_checker.rb

```diff
@@ -86,6 +88,11 @@ def find_latest_resolvable_version
 
       def handle_subprocess_error(error)
         if RESOLVABILITY_ERROR_REGEXES.any? { |rgx| error.message =~ rgx }
+          if error.message.start_with?("module github.com/")
```

Comment:
Should we even care about enterprise at this point? Probably YAGNI for now

Reply:
AFAICT the only hook we have for that is the `api_endpoint` for octokit?: https://github.com/dependabot/dependabot-core/blob/05649b25af72d406b144965b6498167351547081/common/lib/dependabot/clients/github_with_retries.rb#L38

Getting the `Dependabot::Source` out of the `Dependabot::Dependency` that's passed here [seems complicated](https://github.com/dependabot/dependabot-core/blob/05649b25af72d406b144965b6498167351547081/common/lib/dependabot/git_commit_checker.rb#L233-L249), converting the API endpoint to the module path is another problem.

I'm also not sure about assuming GHES customers include the server URL in included module paths? 🤷 
For Enterprise-y reasons, I may want to host my server at `github.acme.com`, but namespace my modules as `acme.com/**`?

Definitely YAGNI.

Reply:
This looks good for now, but I'm wondering the implications for modules hosted on GitHub but have a different module name by using a proxy.  We're probably going to found out once we start using this in production.